vdslib/src/main/java/com/yahoo/vdslib/distribution/Distribution.java

```diff
@@ -123,14 +159,20 @@ public Distribution(String configId) {
     }
 
     public Distribution(StorDistributionConfig config) {
-        int mask = 0;
-        for (int i=0; i<=64; ++i) {
-            distributionBitMasks[i] = mask;
-            mask = (mask << 1) | 1;
-        }
         configSubscriber.configure(config);
     }
 
+    public Distribution(DistributionConfig.Cluster config) {
+        configure(config);
+    }
+
+    private long lastNBits(long value, int n) {
+        if (n < 0 || n > 64)
+            throw new IllegalArgumentException("n must be in [0, 64], but was " + n);
+
+        return value & ((1L << n) - 1);
```

Comment:
If `n == 64` you will get a mask of 0 (since you're really getting `((1L << (n & 0x3F)) - 1) ==> ((1L << 0) - 1)`), which is probably unexpected.

Consider bounding `n` to `[0, 63]` if desired. Note that you should never get `n > 58` in practice for this function.

Reply:
Gosh darnit, I didn't know this. Tack så hemskt mycket!